File: src/main/java/com/sothawo/akkabatch/DoWorkInfo.java
/**
 * $Id: x $
 *
 * Copyright (c) 2014 Jaroso GmbH
 *
 * http://www.jaroso.de
 *
 */
package com.sothawo.akkabatch;

import com.sothawo.akkabatch.messages.DoWork;

/**
 * Information über eine DoWork Nachricht in der Verarbeitung.
 *
 * @author P.J.Meisch (pj.meisch@jaroso.de)
 */
public class DoWorkInfo {
// ------------------------------ FIELDS ------------------------------

    /** die DoWork Nachricht */
    private final DoWork doWork;
    /** Timestamp des letzten Sendevorgangs */
    private long timestamp;
    /** Anzahl bisheriger Sendeversuche */
    private int sendCount;

// --------------------------- CONSTRUCTORS ---------------------------

    public DoWorkInfo(DoWork doWork) {
        this.doWork = doWork;
        this.timestamp = System.currentTimeMillis();
        this.sendCount = 1;
    }

// --------------------- GETTER / SETTER METHODS ---------------------

    public long getTimestamp() {
        return timestamp;
    }
}

File: src/main/scala/com/sothawo/akkabatch/scala/DoWorkInfo.scala
package com.sothawo.akkabatch.scala

import com.sothawo.akkabatch.scala.messages.DoWork
import scala.beans.BeanProperty

/**
 * Information about a DoWork message during processing.
 *
 * @param doWork the DoWork message
 */
class DoWorkInfo(val doWork: DoWork) {

  /** Timestamp of the last send */
  var timestamp = System.currentTimeMillis()

  /** number of sending operations */
  var sendCount = 1

  /**
   * sets the internal timestamp value and increases the sendcount value
   *
   * @param timestamp new timestamp value
   */
  def markResend(timestamp: Long) {
    this.timestamp = timestamp
    sendCount += 1
  }
}
